packages/vite/src/node/plugins/css.ts

```diff
@@ -2023,3 +2090,162 @@ const preProcessors = Object.freeze({
 function isPreProcessor(lang: any): lang is PreprocessLang {
   return lang && lang in preProcessors
 }
+
+const importLightningCSS = createCachedImport(() => import('lightningcss'))
+
+async function compileLightningCSS(
+  id: string,
+  src: string,
+  config: ResolvedConfig,
+  cssConfig: ResolvedLightningCSSOptions,
+  urlReplacer?: CssUrlReplacer,
+): ReturnType<typeof compileCSS> {
+  const deps = new Set<string>()
+  const res = styleAttrRE.test(id)
+    ? (await importLightningCSS()).transformStyleAttribute({
+        filename: id,
+        code: Buffer.from(src),
+        targets: cssConfig.targets,
+        minify: config.isProduction && config.build.cssMinify,
+        analyzeDependencies: true,
+      })
+    : await (
+        await importLightningCSS()
+      ).bundleAsync({
+        filename: id,
+        resolver: {
+          read(filename) {
+            if (filename === id) {
+              // TODO: add source map inline comment
+              return src
+            }
+            return fs.readFileSync(filename, 'utf-8')
+          },
+          async resolve(id, from) {
+            const publicFile = checkPublicFile(id, config)
+            if (publicFile) {
+              return publicFile
+            }
+
+            const resolved = await getAtImportResolvers(config).css(id, from)
+
+            if (resolved) {
+              deps.add(resolved)
+              return resolved
+            }
+            return id
+          },
+        },
+        targets: cssConfig.targets,
+        minify: config.isProduction && config.build.cssMinify,
+        sourceMap: true,
+        analyzeDependencies: true,
+        cssModules: cssModuleRE.test(id),
+        drafts: { nesting: true },
```

Comment:
I'm unsure about enabling this by default yet. The spec is still changing a bit and I wouldn't want to ship it and then have to make a breaking change when the syntax is updated again.

Should there be a way to pass custom Lightning CSS options via the Vite config?

Reply:
Yep I see, I will make `drafts` part of the options and say in the documentation that this is following the spec and can have braking changes